File: inst/examples/e2e-test/R/actionButtons.R
actionButtonsUI <- function(id) {
  ns <- NS(id)
  tagList(
    h1("Action buttons"),
    p("When the button is first rendered, no event is generated - exactly like with actionButton"),
    h4("Click any button to increment the counter."),
    Stack(
      horizontal = TRUE,
      tokens = list(childrenGap = 20),
      DefaultButton.shinyInput(ns("defaultButton"), text = "Default Button"),
      PrimaryButton.shinyInput(ns("primaryButton"), text = "Primary Button"),
      CompoundButton.shinyInput(
        ns("compoundButton"),
        text = "Compound Button",
        secondaryText = "Compound Button has additional text"
      ),
      ActionButton.shinyInput(
        ns("actionButton"),
        text = "Action Button",
        iconProps = list("iconName" = 'AddFriend')
      ),
      CommandBarButton.shinyInput(ns("commandBarButton"), text = "Command Bar Button"),
      CommandButton.shinyInput(ns("commandButton"), text = "Command Button"),
      IconButton.shinyInput(ns("iconButton"), iconProps = list("iconName" = "Table"))
    ),
    textOutput(ns("clicks")),
    actionButton(ns("update"), "Update buttons")
  )
}

actionButtonsServer <- function(id) {
  moduleServer(id, function(input, output, session) {
    ns <- session$ns
    nClicks <- reactiveVal(0)
    
    observeEvent(input$defaultButton, nClicks(nClicks() + 1))
    observeEvent(input$primaryButton, nClicks(nClicks() + 1))
    observeEvent(input$compoundButton, nClicks(nClicks() + 1))
    observeEvent(input$actionButton, nClicks(nClicks() + 1))
    observeEvent(input$commandBarButton, nClicks(nClicks() + 1))
    observeEvent(input$commandButton, nClicks(nClicks() + 1))
    observeEvent(input$iconButton, nClicks(nClicks() + 1))
    
    output$clicks <- renderText({
      sprintf("Number of clicks: %d", nClicks())
    })
    
    observeEvent(input$update, {
      updateDefaultButton.shinyInput(session, "defaultButton", text = "Default Button Updated")
      updatePrimaryButton.shinyInput(session, "primaryButton", text = "Primary Button Updated")
      updateCompoundButton.shinyInput(
        session,
        "compoundButton",
        text = "Compound Button Updated",
        secondaryText = "Compound Button additional text updated"
      )
      updateActionButton.shinyInput(
        session,
        "actionButton",
        text = "Action Button Disabled",
        iconProps = list("iconName"= "Download"),
        disabled = TRUE
      )
      updateCommandBarButton.shinyInput(session, "commandBarButton", text = "Command Bar Button Updated")
      updateCommandButton.shinyInput(session, "commandButton", text = "Command Button Updated")
      updateIconButton.shinyInput(session, "iconButton", iconProps = list("iconName" = "Download"))
    })
  })
}

File: js/cypress/integration/actionButtons.js
function test() {
  let clicks = 1;
  return (name) => {
    cy.get(`#actionButtons-${name}`).click();
    cy.get('#actionButtons-clicks').should('contain', `Number of clicks: ${clicks}`);
    clicks += 1;
  };
}
const testButtonClick = test();

describe('DefaultButton.shinyInput()', () => {
  before(() => {
    cy.visit('/');
  });

  it('click works', () => {
    testButtonClick('defaultButton');
  });
});

describe('PrimaryButton.shinyInput()', () => {
  it('click works', () => {
    testButtonClick('primaryButton');
  });
});

describe('CompoundButton.shinyInput()', () => {
  it('click works', () => {
    testButtonClick('compoundButton');
  });
});

describe('ActionButton.shinyInput()', () => {
  it('click works', () => {
    testButtonClick('actionButton');
  });
});

describe('CommandBarButton.shinyInput()', () => {
  it('click works', () => {
    testButtonClick('actionButton');
  });
});

describe('CommandButton.shinyInput()', () => {
  it('click works', () => {
    testButtonClick('actionButton');
  });
});

describe('IconButton.shinyInput()', () => {
  it('click works', () => {
    testButtonClick('actionButton');
  });
});

describe('Updating buttons', () => {
  before(() => {
    cy.visit('/');
    cy.get('#actionButtons-update').click();
  });

  it('DefaultButton.shinyInput() works', () => {
    cy.get('#actionButtons-defaultButton').within(() => {
      cy.get('.ms-Button-label').should('contain', 'Default Button Updated');
    });
  });

  it('PrimaryButton.shinyInput() works', () => {
    cy.get('#actionButtons-primaryButton').within(() => {
      cy.get('.ms-Button-label').should('contain', 'Primary Button Updated');
    });
  });

  it('CompoundButton.shinyInput() works', () => {
    cy.get('#actionButtons-compoundButton').within(() => {
      cy.get('.ms-Button-label').should('contain', 'Compound Button Updated');
      cy.get('.ms-Button-description').should('contain', 'Compound Button additional text updated');
    });
  });

  it('ActionButton.shinyInput() works', () => {
    cy.get('#actionButtons-actionButton').within(() => {
      cy.get('.ms-Button-label').should('contain', 'Action Button Disabled');
      cy.get('.ms-Icon').should('have.attr', 'data-icon-name', 'Download');
    });
    cy.get('#actionButtons-actionButton').should('have.attr', 'aria-disabled', 'true');
  });

  it('CommandBarButton.shinyInput() works', () => {
    cy.get('#actionButtons-commandBarButton').within(() => {
      cy.get('.ms-Button-label').should('contain', 'Command Bar Button Updated');
    });
  });

  it('CommandButton.shinyInput() works', () => {
    cy.get('#actionButtons-commandButton').within(() => {
      cy.get('.ms-Button-label').should('contain', 'Command Button Updated');
    });
  });

  it('IconButton.shinyInput() works', () => {
    cy.get('#actionButtons-iconButton').within(() => {
      cy.get('.ms-Icon').should('have.attr', 'data-icon-name', 'Download');
    });
  });
});
